### PR TITLE
[FSTORE-328][APPEND] throw intelligible error message if user set primary and partition keys doesn't exist

### PR DESCRIPTION
**Pull request by davitbzh**
This PR fixes bug related to key attribute verification for ExternalFeatureGroups 

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [x] Unit Tests
- [ ] Integration Tests
- [ ] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hsfs/util.py [hunk 1]
```diff
@@ -233,33 +233,36 @@ def get_hostname_replaced_url(sub_path: str):
     return url_parsed.geturl()
 
 
-def verify_attribute_key_names(feature_group_obj):
+def verify_attribute_key_names(feature_group_obj, external_feature_group=False):
```
Comment (moritzmeister) [inline python/hsfs/util.py:236]: This is unnecessary, you can just use `isinstance()` on the feature group object to check if it's an external feature group.

**Review by moritzmeister — CHANGES_REQUESTED** [review]

**Review by moritzmeister — COMMENTED** [review]
How come you don't need to distinguish between external and normal in the java part?

File: python/hsfs/util.py (hunk 1 above)
Comment (davitbzh) [reply python/hsfs/util.py:236]: I did that :) but importing FG in util class introduces circular import error

**Comment by davitbzh** [comment]
> How come you don't need to distinguish between external and normal in the java part?

In Java we provide partition and precombine key names separately (I don´t remember what was the reason for it though). and then setting those values as null in externalFeatureGroupEngine.

**Review by moritzmeister — APPROVED** [review]

**Pull request merged**